lib/shopify_api/collection_pagination.rb

```diff
@@ -0,0 +1,52 @@
+module ShopifyAPI
+  module CollectionPagination
+
+    def next_page?
+      next_page_info.present?
+    end
+
+    def previous_page?
+      previous_page_info.present?
+    end
+
+    def fetch_next_page
+      fetch_page(next_page_info)
+    end
```

Comment:
Thought about it, but decided to not memoize the next/previous page. If you're calling fetch, just have it fetch.